prometheus_client/core.py

```diff
@@ -376,19 +381,26 @@ def _init_value(self, key):
 
         # Update how much space we've used.
         self._used += len(value)
-        struct.pack_into(b'i', self._m, 0, self._used)
+        pack_integer(self._m, 0, self._used)
         self._positions[key] = self._used - 8
 
     def _read_all_values(self):
         """Yield (key, value, pos). No locking is performed."""
+
         pos = 8
-        while pos < self._used:
-            encoded_len = struct.unpack_from(b'i', self._m, pos)[0]
+
+        # performance tricks
```

Comment:
Explain what you're doing a bit more. Not everyone will have seen the relevant talks by Raymond Hettinger

Reply:
https://wiki.python.org/moin/PythonSpeed/PerformanceTips

Reply:
I know why this works, but someone reading this code might not. A quick comment will help future readers of the code.

Reply:
fixed